### PR TITLE
[Backport] Fix bug when migrate from API (#8631)

### PR DESCRIPTION
**Pull request by 6543**
backport #8631

### CODE REVIEW

**Review by techknowlogick — APPROVED** [review]

File: routers/api/v1/repo/repo.go [hunk 1]
```diff
@@ -433,17 +433,32 @@ func Migrate(ctx *context.APIContext, form auth.MigrateRepoForm) {
...
+		return
+	}
+
+	opts.MigrateToRepoID = repo.ID
+
 	defer func() {
 		if e := recover(); e != nil {
 			var buf bytes.Buffer
 			fmt.Fprintf(&buf, "Handler crashed with error: %v", log.Stack(2))
+
```
Comment (6543) [inline routers/api/v1/repo/repo.go]:
```suggestion
```

**Comment by lunny** [comment]
Maybe another of my PRs on v1.10.2 fixed this? #9511

**Comment by 6543** [comment]
@lunny I just created a backport ... should I close it

File: models/repo.go [hunk 2]
```diff
@@ -2811,3 +2811,25 @@ func (repo *Repository) GetOriginalURLHostname() string {
 
 	return u.Host
 }
+
+// GetTreePathLock returns LSF lock for the treePath
+func (repo *Repository) GetTreePathLock(treePath string) (*LFSLock, error) {
```
Comment (zeripath) [inline models/repo.go]: ? unrelated?

Comment (6543) [reply models/repo.go]: Coul be generated ba cherry-pick + conflict resolve ...

Comment (6543) [reply models/repo.go]: done

**Comment by 6543** [comment]
@zeripath done

**Comment by 6543** [comment]
@lunny can this go in?

**Review by lafriks — APPROVED** [review]
